packages/server/utils/ServerHealthChecker.ts

```diff
@@ -8,45 +8,41 @@ import sendToSentry from './sendToSentry'
 
 const REDIS_URL = process.env.REDIS_URL!
 const SERVER_ID = process.env.SERVER_ID!
+const INSTANCE_ID = `${SERVER_ID}:${process.pid}`
 
 export default class ServerHealthChecker {
-  publisher = new Redis(REDIS_URL, {connectionName: 'serverHealth_pub'})
-  subscriber = new Redis(REDIS_URL, {connectionName: 'serverHealth_sub'})
-  pendingPongs: null | Set<string> = null
+  private publisher = new Redis(REDIS_URL, {connectionName: 'serverHealth_pub'})
+  private subscriber = new Redis(REDIS_URL, {connectionName: 'serverHealth_sub'})
+  private remoteSocketServers: null | string[] = null
   constructor() {
     this.subscriber.on('message', (channel, remoteServerId) => {
       if (channel === 'socketServerPing') {
-        if (remoteServerId === SERVER_ID) return
-        this.publisher.publish(`socketServerPong:${remoteServerId}`, SERVER_ID)
-      } else if (channel === `socketServerPong:${SERVER_ID}`) {
-        this.pendingPongs?.delete(remoteServerId)
+        if (remoteServerId === INSTANCE_ID) return
+        this.publisher.publish(`socketServerPong:${remoteServerId}`, INSTANCE_ID)
+      } else if (channel === `socketServerPong:${INSTANCE_ID}`) {
+        if (!this.remoteSocketServers) {
+          console.error('unsolicited pong received before getLivingServers was called')
+        } else {
+          this.remoteSocketServers.push(remoteServerId)
+        }
       }
     })
-    this.subscriber.subscribe('socketServerPing', `socketServerPong:${SERVER_ID}`)
-    this.publisher.sadd('socketServers', SERVER_ID)
+    this.subscriber.subscribe('socketServerPing', `socketServerPong:${INSTANCE_ID}`)
```

Comment:
-1 I did some testing by running `yarn dev` and then running `SERVER_ID=9 node scripts/runSocketServer.js`. It turns out this would clear the presence completely. After some digging, this is the culprit. We need to await the `subscribe` call before doing any pings.

Reply:
i can't reproduce this locally, but i believe you!

while debugging i also found out it was possible to double subscribe to messages, so i made it a singleton.
Also, since we've never downscaled the number of servers we have, i got rid of calling this when the server goes down & only when they come up. let me know what you think!